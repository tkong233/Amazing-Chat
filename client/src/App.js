import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import jwt_decode from "jwt-decode";
import setAuthToken from "./utils/setAuthToken";

import { setCurrentUser, logoutUser } from "./actions/authActions";
import { PersistGate } from 'redux-persist/integration/react'
import { Provider } from "react-redux";
import {store, persistor} from "./store";

import Navbar from "./components/layout/Navbar";
import Landing from "./components/layout/Landing";
import Register from "./components/auth/Register";
import Login from "./components/auth/Login";
import Reset from "./components/auth/Reset";
import Profile from "./components/Profile";
import PrivateRoute from "./components/private-route/PrivateRoute";
import Dashboard from "./components/dashboard/Dashboard";

import GlobalStyles from './components/global/GlobalStyles';

import "./App.css";

// Check for token to keep user logged in
if (localStorage.jwtToken) {
  // Set auth token header auth
  const token = localStorage.jwtToken;
  setAuthToken(token);
  // Decode token and get user info and exp
  const decoded = jwt_decode(token);
  // Set user and isAuthenticated
  store.dispatch(setCurrentUser(decoded));
  // Check for expired token
  const currentTime = Date.now() / 1000; // to get in milliseconds
  if (decoded.exp < currentTime) {
    // Logout user
    store.dispatch(logoutUser());

    // Redirect to login
    window.location.href = "./login";
  }
}
class App extends Component {
  render() {
    return (
      <div>
      <GlobalStyles />
      <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Router>
          <div className="App">
            <Navbar />
            <Route exact path="/" component={Landing} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/reset" component={Reset} />
            {/* <Route exact path="/profile" component={Profile} /> */}
            <Switch>
              <PrivateRoute exact path="/dashboard" component={Dashboard} />
              <PrivateRoute exact path="/profile" component={Profile} />
            </Switch>
          </div>
        </Router>
        </PersistGate>
      </Provider>
      </div>
    );
  }
}
export default App;
