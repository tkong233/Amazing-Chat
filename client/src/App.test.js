import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

describe('App Component', ()=>{
  describe('Renders', ()=>{
      let wrapper;
      beforeEach(()=>{
          
          wrapper = shallow(<App />);
      });
      
      it('Should render App', ()=>{
          const app = wrapper.find(`[data-test='AppComponent']`);
          expect(app.length).toBe(1);
      });

      it('Snapshot testing', ()=>{
          expect(wrapper).toMatchSnapshot();
      });
      

  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});
