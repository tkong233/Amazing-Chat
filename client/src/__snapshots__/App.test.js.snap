// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
